### PR TITLE
tests: Add Sink API E2E tests

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-596

All tests will pass in the final [PR](https://github.com/keboola/keboola-as-code/pull/1770).

[Large set of fixes and E2e tests](https://github.com/keboola/keboola-as-code/pull/1759) is divided into smaller PRs to do a review.

--------------

**Changes:**
- Added Sink API E2E tests.

-----------

Commend per commits. Please, explore [the final version](https://github.com/keboola/keboola-as-code/pull/1759) in the IDE.

-------------

### CODE REVIEW

File: test/stream/api/sink/statistics-total-001-ok/002-poll-source/request.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "path": "<<001-create-source:response.url>>",
+  "method": "GET",
+  "headers": {
+    "X-StorageApi-Token": "%%TEST_KBC_STORAGE_API_TOKEN%%"
+  },
+  "repeat": {
+    "until": "status != 'processing'",
+    "timeout": 60
+  }
```
Comment (Matovidlo) [inline test/stream/api/sink/statistics-total-001-ok/002-poll-source/request.json:10]: This repeats request on `stream API`  till timeout of status is not `processing` something like `testify.Eventually()`?

File: test/stream/api/sink/create-011-exists-409/004-poll-sink/request.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (Matovidlo) [inline test/stream/api/sink/create-011-exists-409/004-poll-sink/request.json:1]: Do we need to poll sink if we just want to return the colision `409`? Or is it different use case?

File: test/stream/api/sink/delete-001-ok/expected-etcd-kvs.txt [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+>>>>>
+
+<<<<<
+definition/source/version/%%TEST_KBC_PROJECT_ID%%/%%TEST_DEFAULT_BRANCH_ID%%/my-source/0000000001
+-----
+%A
+>>>>>
+
+<<<<<
+task/%%TEST_KBC_PROJECT_ID%%/api.create.source/%%TEST_DEFAULT_BRANCH_ID%%/my-source/%s
```
Comment (Matovidlo) [inline test/stream/api/sink/delete-001-ok/expected-etcd-kvs.txt:32]: What `task` represents?

File: test/stream/api/sink/list-001-empty-ok/003-list-sink/expected-response.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  "projectId": %%TEST_KBC_PROJECT_ID%%,
+  "branchId": %%TEST_DEFAULT_BRANCH_ID%%,
+  "sourceId": "my-source",
+  "page": {
+    "limit": 100,
+    "totalCount": 0,
+    "sinceId": "",
+    "lastId": ""
+  },
+  "sinks": []
```
Comment (Matovidlo) [inline test/stream/api/sink/list-001-empty-ok/003-list-sink/expected-response.json:11]: Why is `sink` empty if we said that is always have `file` and `slice` prealocated? Or did I missed something?

File: test/stream/api/sink/list-002-ok/009-list-limit-2-page-004/request.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "path": "/v1/branches/%%TEST_DEFAULT_BRANCH_ID%%/sources/my-source/sinks?limit=2&sinceId=my-sink-5",
```
Comment (Matovidlo) [inline test/stream/api/sink/list-002-ok/009-list-limit-2-page-004/request.json:2]: If I use `sinceId=my-sink-4` I sould receive `my-sink-4` in response. Am I right?

**Review by hosekpeter — APPROVED** [review]

File: test/stream/api/sink/list-002-ok/009-list-limit-2-page-004/request.json (hunk 5 above)
Comment (michaljurecko) [reply test/stream/api/sink/list-002-ok/009-list-limit-2-page-004/request.json:2]: You can rename the parameter in the URL to `afterId`, if you want.

Comment (michaljurecko) [reply test/stream/api/sink/list-002-ok/009-list-limit-2-page-004/request.json:2]: @jachym-tousek-keboola please create a task for the renaming

**Pull request merged**